libyul/optimiser/NameDispenser.cpp

```diff
@@ -61,3 +62,9 @@ bool NameDispenser::illegalName(YulString _name)
 {
 	return isRestrictedIdentifier(m_dialect, _name) || m_usedNames.count(_name);
 }
+
+void NameDispenser::reset(Block const& _ast)
```

Comment:
I think instead of adding a `reset` function it might be more idiomatic to re-assign `suite.m_dispenser` from a fresh object instead.

Reply:
I did think about that, but `NameDispenser` is not really copyable/assignable because of the member `Dialect const& m_dialect`. Not sure if changing it to a pointer now is worth the hassle. What do you think?

Reply:
We could make OptimiserSuite::m_dispenser a unique_ptr but keep OptiimserStepContext::dispenser a reference. This should require few changes, does it?

Reply:
> We could make OptimiserSuite::m_dispenser a unique_ptr but keep OptiimserStepContext::dispenser a reference. This should require few changes, does it?

No, we have to make both of them into a pointer. Because the problem here is that the steps need to use the reference `OptimiserStepContext::dispenser`. So if we assign a new Dispenser to the `unique_ptr OptimizerSuite::m_dispenser`, then the reference should also be changed. So, we'll have to make both of them into a pointer.  I'd prefer a reset function over making this change.

Reply:
Oh right, of course!